components/org.wso2.carbon.appmgt.mobile/src/main/java/org/wso2/mobile/utils/utilities/ZipFileReading.java

```diff
@@ -109,8 +109,14 @@ public String readiOSManifestFile(String filePath, String name) {
                         break;
                     }
                 }
-                NSDictionary rootDict = (NSDictionary) BinaryPropertyListParser
-                        .parse(buffer.toByteArray());
+                NSDictionary rootDict;
+                try{
+                    rootDict = (NSDictionary) BinaryPropertyListParser
+                            .parse(buffer.toByteArray());
+                }catch(IllegalArgumentException e){
+                    log.debug("Uploaded file didn't have a Binary Plist");
```

Comment:
Looks like debug enabled check is missing. Even though the impact is minimal in this particular case as no expensive logic i.e. String concatination, etc gets executed, sufficient amount of attention should always be paid upon following best practices.